pkg/doc/jose/jwe.go

```diff
@@ -334,28 +334,38 @@ func deserializeAndDecodeHeaders(rawJWE *rawJSONWebEncryption) (*Headers, *Heade
 	return &protectedHeaders, &unprotectedHeaders, nil
 }
 
-func deserializeRecipients(rawJWE *rawJSONWebEncryption) ([]*Recipient, error) {
-	var recipients []*Recipient
+func parseDeserializeRecipients(rawJWE *rawJSONWebEncryption) ([]*Recipient, error) {
```

Comment:
@Baha-sk fyi

Reply:
This refactoring is ok, if rawJWE.Recipient is not empty, then an array of recipients will returned, otherwise a single recipient is assumed and we retrieve recipient info from protected headers and `rawJWE.B64SingleRecipientEncKey` which is repopulated by Tink's ecdhes/ecdh1pu primitives if a single recipient was used.

The only improvement was my other comment below, it can be added later after this PR. It's an additional preventive check, but the crypto primitives properly generate the ciphertext that is populating `rawJWE` with all the necessary fields needed here.